test/aggregator_test.rb

```diff
@@ -139,4 +159,89 @@ def test_with_prefix
     assert_equal("foo", @sink.datagrams.last.name)
     assert_equal(1, @sink.datagrams.last.value)
   end
+
+  def test_synchronous_operation_on_thread_failure
+    # Force thread_healthcheck to return false
+    @subject.stubs(:thread_healthcheck).returns(false)
+
+    @subject.increment("foo", 1, tags: { foo: "bar" })
+    @subject.aggregate_timing("bar", 100, tags: { foo: "bar" })
+    @subject.gauge("baz", 100, tags: { foo: "bar" })
+
+    # Verify metrics were sent immediately
+    assert_equal(3, @sink.datagrams.size)
```

Comment:
This might be paranoid, but perhaps it's also worth asserting that no methods were called on the `Aggregator`'s internal `@aggregation_state`. Otherwise the test might be a little racy, I'm not sure.

I may be overthinking that. 🤷

Reply:
Added assertions to be on the safe side.